compass/ocean/tests/global_convergence/cosine_bell/__init__.py

```diff
@@ -0,0 +1,81 @@
+from compass.testcase import TestCase
+
+from compass.ocean.tests.global_convergence.cosine_bell.mesh import Mesh
+from compass.ocean.tests.global_convergence.cosine_bell.init import Init
+from compass.ocean.tests.global_convergence.cosine_bell.forward import Forward
+from compass.ocean.tests.global_convergence.cosine_bell.analysis import \
+    Analysis
+
+
+class CosineBell(TestCase):
+    """
+    A test case for creating a global MPAS-Ocean mesh
+
+    Attributes
+    ----------
+    resolutions : list of int
+    """
+    def __init__(self, test_group):
+        """
+        Create test case for creating a global MPAS-Ocean mesh
+
+        Parameters
+        ----------
+        test_group : compass.ocean.tests.cosine_bell.GlobalOcean
+            The global ocean test group that this test case belongs to
+        """
+        super().__init__(test_group=test_group, name='cosine_bell')
+        self.resolutions = None
+
+    def configure(self):
+        """
+        Set config options for the test case
+        """
+        config = self.config
+        resolutions = config.get('cosine_bell', 'resolutions')
+        resolutions = [int(resolution) for resolution in
+                       resolutions.replace(',', ' ').split()]
+
+        # dt is proportional to resolution: default 30 seconds per km
+        dt_per_km = config.getint('cosine_bell', 'dt_per_km')
+
+        self.resolutions = resolutions
+
+        for resolution in resolutions:
+            self.add_step(Mesh(test_case=self, resolution=resolution))
+            self.add_step(Init(test_case=self, resolution=resolution))
+            self.add_step(Forward(test_case=self, resolution=resolution,
+                                  dt=dt_per_km*resolution))
+        self.add_step(Analysis(test_case=self, resolutions=resolutions))
+
+        self.update_cores()
+
+    def run(self):
+        """
+        Run each step of the testcase
+        """
+        self.update_cores()
+
+        # run the step
+        super().run()
+
+    def update_cores(self):
+        """ Update the number of cores and min_cores for each forward step """
+
+        config = self.config
+
+        goal_cells_per_core = config.getfloat('cosine_bell',
+                                              'goal_cells_per_core')
+        max_cells_per_core = config.getfloat('cosine_bell',
+                                             'max_cells_per_core')
+
+        for resolution in self.resolutions:
+            # a heuristic based on QU30 (65275 cells) and QU240 (10383 cells)
+            approx_cells = 6e8 / resolution**2
+            # ideally, about 300 cells per core
+            cores = max(1, int(approx_cells / goal_cells_per_core + 0.5))
+            # In a pinch, about 3000 cells per core
+            min_cores = max(1, int(approx_cells / max_cells_per_core + 0.5))
```

Comment:
@vanroekel, let me know if you're okay with this heuristic or if you prefer to hard-code cores (and min. cores) for each resolution.  I thought this would be nice because it adds flexibility.